src/main/ruby/core/kernel.rb

```diff
@@ -76,7 +76,7 @@ def Float(obj)
   module_function :Float
 
   def Hash(obj)
-    return {} if obj.nil? || obj == []
+    return {} if obj.equal?(nil) || obj == []
```

Comment:
I wonder, what was the problem here with `nil?`?
Is it that obj can be a BasicObject?

Reply:
Yes, `BasicObject` is used in these specs.